tests/test_cliCmdQueryDatasets.py

```diff
@@ -357,6 +357,13 @@ def test_limit_order(self):
         ]
         self.assertAstropyTablesEqual(tables, expectedTables, filterColumns=True)
 
+        # Same as previous test, but with positive limit so no warning is
+        # issued.
+        tables = self._queryDatasets(
+            repo=testRepo.butler, limit=1, order_by=("visit"), collections="*", glob="*"
```

Comment:
Maybe test that no warning was issued (by issuing your own warning and then making sure you only got the one warning)?